packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -42,17 +42,32 @@ def _read_in_chunks(file_object, chunk_size=1024 * 8):
         yield data
 
 
+class _FastZipFileReader(zipfile.ZipFile):
+    """Internal used to gain speed while unzipping many files"""
+
+    def _RealGetContents(self):
```

Comment:
This looks dangerous at first sight. Can you please add some reference or test that clarifies a bit more overriding an internal member  and "emptying" it?
 
Is there a test that will catch any change on the implementation of this part of ``zipfile``

Reply:
There are already tests in place to test if this works properly. Please have a look at the test suite.

Let me know if the refactored doctoring are better now.